rasa/nlu/featurizers/sparse_featurizer/regex_featurizer.py

```diff
@@ -259,10 +184,10 @@ def _features_for_patterns(
 
         sequence_length = len(tokens)
 
-        max_number_patterns = self.vocabulary_stats["max_number_patterns"]
+        num_patterns = len(self.known_patterns)
 
-        sequence_features = np.zeros([sequence_length, max_number_patterns])
-        sentence_features = np.zeros([1, max_number_patterns])
+        sequence_features = np.zeros([sequence_length, num_patterns])
+        sentence_features = np.zeros([1, num_patterns])
```

Comment:
Please ensure that the order of old patterns is not disturbed. And add a test for it.